nifi-extension-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/GetSlackReaction.java

```diff
@@ -0,0 +1,294 @@
+            .name("failure")
+            .description("FlowFiles are routed to this relationship if an error occurred when fetching reactions")
+            .build();
+
+    static final Relationship REL_WAIT = new Relationship.Builder()
+            .name("wait")
+            .description("Self loop relationship which is used when waiting for further reactions")
+            .build();
+
+    static final Relationship REL_NO_REACTION = new Relationship.Builder()
+            .name("no_reaction")
+            .description("FlowFiles are routed to this relationship if no reaction arrived for the message in the given timeframe")
+            .build();
+
+    private static final List<PropertyDescriptor> DESCRIPTORS = List.of(CHANNEL_ID,
+            THREAD_TIMESTAMP,
+            ACCESS_TOKEN,
+            WAIT_MONITOR_WINDOW,
+            RELEASE_IF_ONE_REACTION);
+
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_SUCCESS,
+            REL_WAIT,
+            REL_NO_REACTION,
+            REL_FAILURE);
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    private RateLimit rateLimit;
+    private volatile App slackApp;
+    private MethodsClient client;
+
+    @OnScheduled
+    public void setup(final ProcessContext context) {
+        rateLimit = new RateLimit(getLogger());
+        slackApp = createSlackApp(context);
+        client = initializeClient(slackApp);
+    }
+
+    @OnStopped
+    public void shutdown() {
+        if (slackApp != null) {
+            slackApp.stop();
+            slackApp = null;
+        }
+        rateLimit = null;
+    }
+
+    public RateLimit getRateLimit() {
+        return rateLimit;
+    }
+
+    private App createSlackApp(final ProcessContext context) {
+        final String botToken = context.getProperty(ACCESS_TOKEN).getValue();
+        final AppConfig appConfig = AppConfig.builder()
+                .singleTeamBotToken(botToken)
+                .build();
+
+        return new App(appConfig);
+    }
+
+    protected MethodsClient initializeClient(final App slackApp) {
+        slackApp.start();
+        return slackApp.client();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        if (rateLimit.isLimitReached()) {
+            getLogger().debug("Will not fetch reactions from Slack because rate limit has been reached");
+            context.yield();
+            return;
+        }
+
+        FlowFile flowFile = session.get();
+
+        if (flowFile == null) {
+            return;
+        }
+
+        final String botToken = context.getProperty(ACCESS_TOKEN).getValue();
+        final String channelId = context.getProperty(CHANNEL_ID).evaluateAttributeExpressions(flowFile).getValue();
```

Comment:
I dont think the user supplied processor property value is a valid use case. 

Also "Channel ID" doesn't seem valid as the flowfile attribute to look up as nothing preceding this component would set that.    ConsumeSlack for instance sets "slack.channel.id"